### DISCUSSION TITLE
View and edit custom fields in issues

### DISCUSSION THREAD

**Discussion by hulkabob** [ankitpokhrel/jira-cli#375] [Ideas] [answered]
Often we have somebody file in a request in a jira issue with a custom field, like Story Points or fixVersions not set. 
And we with our team constantly need to have these values in place. so sprint metrics like velocity and burnout charts are in proper shape. 

When it is possible to create an issue with custom fields, we cannot see the actual value of it, neither edit it. 
If such a feature would be sometime available, this would greatly help us with the lot of tickets we need to sort out.
Thanks in advance!

**Comment by ankitpokhrel** [comment, marked as answer]
Hi @hulkabob, support for editing custom field is done in #377 and available from [v1.0.0-beta.1](https://github.com/ankitpokhrel/jira-cli/releases/tag/v1.0.0-beta.1). Please feel free to test and report any issue found.

Regarding custom fields on issue view, that's something I need to think about. There isn't any plan for this at the moment but I will see if I can find any workaround.

**Reply by hulkabob** [reply]
I built my one from git, let me check out if there is such a thing. 
Thanks for the update

**Reply by hulkabob** [reply]
Yup, there is such an option, rebuilt from main branch. 
But there's an issue with updating my value:
```jira issue edit ID-1234 --custom "story-points"=5```
```
Error:
  - customfield_10026: Field 'customfield_10026' cannot be set. It is not on the appropriate screen, or unknown.

jira: Received unexpected response '400 Bad Request'.
Please check the parameters you supplied and try again.
```
Checked from browser - jira actually returns `customfield_10026` for story points

**Reply by ankitpokhrel** [reply]
Hi @hulkabob, You need the field on the edit screen to be able to set it. Are you sure this field is available in the edit screen in the UI? Looks like its not available for the issue you are trying to set story points to.

**Reply by hulkabob** [reply]
Yes, the field is available, got this form data from jira ui on SP edit
```
atl_token:  "47671b57-b2b8-440f-8356-4ad2db290d15_*****************_lin"
customfield_10026:  "5"
issueId: "12996"
singleFieldEdit: "true"
skipScreenCheck:  "true"
```
Tried enabling --debug, but it didn't embed the actual request body, only headers.

**Reply by ankitpokhrel** [reply]
That's strange! The error is from Jira and you only get it if the field is not available on the edit screen. If you see the screenshot below the first one `434` fails because story point field is not available for issue type `Bug`. The other one `460` runs fine as SP field is available for ticket type `Story`. So I suspect this has something to do with your setup.

![Screen Shot 2022-06-01 at 9 01 47 AM](https://user-images.githubusercontent.com/2364546/171346554-ed26f3e4-1b2b-4c9c-b788-c9a1a442ff18.png)

Could you please check if it works when you send request to the Jira API directly? If not, this is something related to your issue type setting.

```sh
 curl --request PUT \
--url 'https://<host>/rest/api/2/issue/<ISS-NR>' \
--user "<user>:${JIRA_API_TOKEN}" \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--data '{
  "update": {"description":[{"set":"API test for custom field edit..."}],"customfield_10026":[{"set":5}]}
}
```

**Reply by hulkabob** [reply]
That is strange ideed. 
```curl --request PUT --url 'https://SOMEJIRA.atlassian.net/rest/api/2/issue/MP-2450' --user "USERNAME:${JIRA_API_TOKEN}" --header 'Accept: application/json' --header 'Content-Type: application/json' --data '{
  "update": {"description":[{"set":"API test for custom field edit..."}],"customfield_10026":[{"set":5}]}
}'
{"errorMessages":[],"errors":{"customfield_10026":"Field 'customfield_10026' cannot be set. It is not on the appropriate screen, or unknown."}}
```
Also tried setting fields from the form, no result. Not completely sure what's happening, maybe our jira setup has something fancy in it's config. 
```
$ curl --request PUT --url 'https://SOMEJIRA.atlassian.net/rest/api/2/issue/MP-2450' --user "USESRNAME:${JIRA_API_TOKEN}" --header 'Accept: application/json' --header 'Content-Type: application/json' --data '{
  "update": {"description":[{"set":"API test for custom field edit..."}],"customfield_10026":[{"set":5}]}, "singleFieldEdit":"true", "skipScreenCheck":"true" }'
{"errorMessages":[],"errors":{"customfield_10026":"Field 'customfield_10026' cannot be set. It is not on the appropriate screen, or unknown."}}
```
Also I had an old installation of go-jira, it fails with the same error. So it's not the tool failing, but the cloud jira being stubborn
